library/psa_crypto.c

```diff
@@ -3659,24 +3656,18 @@ psa_status_t psa_sign_hash( psa_key_handle_t handle,
         goto exit;
     }
 
-#if defined(MBEDTLS_PSA_CRYPTO_SE_C)
-    if( psa_get_se_driver( slot->attr.lifetime, &drv, &drv_context ) )
-    {
-        if( drv->asymmetric == NULL ||
-            drv->asymmetric->p_sign == NULL )
-        {
-            status = PSA_ERROR_NOT_SUPPORTED;
-            goto exit;
-        }
-        status = drv->asymmetric->p_sign( drv_context,
-                                          slot->data.se.slot_number,
-                                          alg,
-                                          hash, hash_length,
-                                          signature, signature_size,
-                                          signature_length );
-    }
-    else
-#endif /* MBEDTLS_PSA_CRYPTO_SE_C */
+    /* Try any of the available accelerators first */
+    status = psa_driver_wrapper_sign_hash( slot,
+                                           alg,
+                                           hash,
+                                           hash_length,
+                                           signature,
+                                           signature_size,
+                                           signature_length );
+    if( status != PSA_ERROR_NOT_SUPPORTED )
+        goto exit;
+
+    /* If the operation was not supported by any accelerator, try fallback. */
```

Comment:
If the key is opaque, the code below this point attempts to access it as a transparent key. This only works if it's guaranteed that the code below isn't called on opaque keys, which in turn relies on the driver wrapper never returning `PSA_ERROR_NOT_SUPPORTED` for an opaque key. This is fragile and it doesn't stand a chance of being caught by static analysis. Please add a check directly in the main function in `psa_crypto.c` so that no matter what the driver wrapper does, the software implementation isn't invoked on an opaque key.